pkg/events/derive/hidden_kernel_module.go

```diff
@@ -25,36 +26,83 @@ import (
 
 var (
 	foundHiddenKernModsCache *lru.Cache
+	wakeupChannel            = make(chan ScanRequest)
 )
 
+const (
+	procModules    uint32 = 1 << 0
+	kset                  = 1 << 1
+	mod_tree              = 1 << 2
+	sendToVerifier        = 1 << 29
+	fullScan              = 1 << 30
+)
+
+type ScanRequest struct {
+	Address  uint64
+	FullScan bool
+}
+
 func HiddenKernelModule() DeriveFunction {
 	return deriveSingleEvent(events.HiddenKernelModule, deriveHiddenKernelModulesArgs())
 }
 
+// deriveHiddenKernelModulesArgs
 func deriveHiddenKernelModulesArgs() deriveArgsFunction {
 	return func(event trace.Event) ([]interface{}, error) {
 		address, err := parse.ArgVal[uint64](&event, "address")
 		if err != nil {
 			return nil, err
 		}
 
-		name, err := parse.ArgVal[[]byte](&event, "name")
+		if _, found := foundHiddenKernModsCache.Get(address); found {
+			return nil, nil // Already reported this event - no need to report again
+		}
+
+		nameBytes, err := parse.ArgVal[[]byte](&event, "name")
 		if err != nil {
 			return nil, err
 		}
+		name := string(nameBytes[:bytes.IndexByte(nameBytes[:], 0)]) // Remove the trailing terminating characters
 
-		if _, found := foundHiddenKernModsCache.Get(address); found {
-			return nil, nil // already reported this event - no need to report again
-		} else {
-			foundHiddenKernModsCache.Add(address, struct{}{}) // so we won't report multiple times
+		srcversionBytes, err := parse.ArgVal[[]byte](&event, "srcversion")
+		if err != nil {
+			return nil, err
+		}
+		srcversion := string(srcversionBytes[:bytes.IndexByte(srcversionBytes[:], 0)]) // Remove the trailing terminating characters
+
+		flags, err := parse.ArgVal[uint32](&event, "flags")
+		if err != nil {
+			return nil, err
 		}
 
+		if flags&procModules != 0 {
```

Comment:
```suggestion
		if flags&procModules {
```

Reply:
ditto for the next `if`s.

Reply:
Without it the compiler yells:
pkg/events/derive/hidden_kernel_module.go:78:6: non-boolean condition in if statement